Common/TableProducer/PID/pidTPCML.h

```diff
@@ -0,0 +1,254 @@
+  - Input:
+    -- modelPath:           std::string   ; Path to the model file;
+    -- enableOptimization:  bool          ; enabling optimizations for the loaded model in the session options;
+  */
+
+  LOG(info) << "--- Neural Network for the TPC PID response correction ---";
+  mEnv = std::make_shared<Ort::Env>(ORT_LOGGING_LEVEL_WARNING, "pid-neural-network");
+  if (enableOptimization) {
+    sessionOptions.SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_EXTENDED);
+  }
+  mSession.reset(new Ort::Experimental::Session{*mEnv, modelPath, sessionOptions});
+
+  mInputNames = mSession->GetInputNames();
+  mInputShapes = mSession->GetInputShapes();
+  mOutputNames = mSession->GetOutputNames();
+  mOutputShapes = mSession->GetOutputShapes();
+
+  LOG(info) << "Input Nodes:";
+  for (size_t i = 0; i < mInputNames.size(); i++) {
+    LOG(info) << "\t" << mInputNames[i] << " : " << printShape(mInputShapes[i]);
+  }
+
+  LOG(info) << "Output Nodes:";
+  for (size_t i = 0; i < mOutputNames.size(); i++) {
+    LOG(info) << "\t" << mOutputNames[i] << " : " << printShape(mOutputShapes[i]);
+  }
+
+  LOG(info) << "--- Network initialized! ---";
+
+} // Network::Network(std::string, bool)
+
+Network& Network::operator=(Network& inst)
+{
+
+  /*
+  Operator: Setting instances of one class to the instances of an input class
+  - Input:
+    -- inst:    const Network&  ; An instance of the network class;
+  - Output:
+    -- *this:   Network&        ; An instance with the private properties of the "inst" (input) instance;
+  */
+
+  mEnv = inst.mEnv;
+  mSession = inst.mSession;
+  // sessionOptions = inst.sessionOptions; // Comment (Christian): Somehow the session options throw an error when copying
+  mInputNames = inst.mInputNames;
+  mInputShapes = inst.mInputShapes;
+  mOutputNames = inst.mOutputNames;
+  mOutputShapes = inst.mOutputShapes;
+
+  LOG(debug) << "Network copied!";
+
+  return *this;
+
+} // Network& Network::operator=(const Network &)
+
+template <typename T>
+std::vector<float> Network::createInputFromTrack(const T& track, const uint8_t id) const
+{
+
+  /*
+  Function: Creating a std::vector<float> from a track with the variables that the network has been trained on
+  - Input:
+    -- track:         const T&            ;   A track, typically from soa::Join<...> tables or of their iterators;
+    -- id:            uint8_t             ;   The id of a particle used for the mass assignment with o2::track::pid_constants::sMasses[id];
+  - Output:
+    -- inputValues:   std::vector<float>  ;   A std::vector<float> with the input variables for the network;
+  */
+
+  const float p = track.tpcInnerParam();
+  const float tgl = track.tgl();
+  const float signed1Pt = track.signed1Pt();
+  const float eta = track.eta();
+  const float mass = o2::track::pid_constants::sMasses[id];
+  const float bg = p / mass;
+  const float ncl = std::sqrt(63. / track.tpcNClsFound());
```

Comment:
This should be const float ncl = std::sqrt(159. / track.tpcNClsFound());

Reply:
Thanks, fixed it now

Reply:
Maybe we can have the 159. taken from some `constexpr` somewhere or defined it explicity for future reference?